### ISSUE TITLE
AppImage never Runs - Linux Mint 21.1

### ISSUE THREAD

**Issue by mooo25** [m4dEngi/RemotePlayWhatever#114]
I downloaded the latest appimage today, and when executing, i hear my computer spin up as if it were running something, but nothing happens.

No program starts in the tray.

Steam is already running when I try to run the appimage.  I have also tried to run the appimage when steam is not running.

**Comment by m4dEngi** [comment]
Yeah, the readme is a bit outdated, the app doesn't use the tray any more. Did it show any errors when run from terminal?

**Comment by mooo25** [comment]
Thanks for the response!No, there was no errors when run off of the desktop. I am a Linux noob, so idk how to run it in the terminal. How would I do that on Mint?
-------- Original message --------From: Dmitry ***@***.***> Date: 9/5/23  6:24 PM  (GMT-05:00) To: m4dEngi/RemotePlayWhatever ***@***.***> Cc: mooo25 ***@***.***>, Author ***@***.***> Subject: Re: [m4dEngi/RemotePlayWhatever] AppImage never Runs - Linux Mint 21.1 (Issue #114) 
Yeah, the readme is a bit outdated, the app doesn't use the tray any more. Did it show any errors when run from terminal?

**Comment by m4dEngi** [comment]
In some DEs you can just start terminal and drag & drop appimage into the terminal window and then hit enter (removing the `file:/` prefix if it gets there too). If the permissions are set right, it should start the app and spit something into stdout/stderr, which will be printed in the terminal window.

**Comment by mooo25** [comment]
Thanks!I'll give it a try and let you know what happens. If I need more help, I'll let you know too. 
-------- Original message --------From: Dmitry ***@***.***> Date: 9/5/23  6:39 PM  (GMT-05:00) To: m4dEngi/RemotePlayWhatever ***@***.***> Cc: mooo25 ***@***.***>, Author ***@***.***> Subject: Re: [m4dEngi/RemotePlayWhatever] AppImage never Runs - Linux Mint 21.1 (Issue #114) 
In some DEs you can just start terminal and drag & drop appimage into the terminal window and then hit enter (removing the file:/ prefix if it gets there too). If the permissions are set right, it should start the app and spit something into stdout/stderr, which will be printed in the terminal window.

**Comment by mooo25** [comment]
Here was the terminal output when I did as you suggested:

`Gtk-Message: 18:13:52.927: Failed to load module "window-decorations-gtk-module"
Gtk-Message: 18:13:52.938: Failed to load module "colorreload-gtk-module"

(remoteplaywhatever:2155): Gtk-WARNING **: 18:13:53.216: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/breeze/status/64/image-missing.svg: Unable to load image-loading module: libpixbufloader-svg.so: libpixbufloader-svg.so: cannot open shared object file: No such file or directory (gdk-pixbuf-error-quark, 5)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /usr/share/icons/breeze/status/64/image-missing.svg: Unable to load image-loading module: libpixbufloader-svg.so: libpixbufloader-svg.so: cannot open shared object file: No such file or directory (gdk-pixbuf-error-quark, 5)
Aborted (core dumped)
`

**Comment by mooo25** [comment]
This was the output when doing as you suggested but with "sudo" in front:

```
(remoteplaywhatever:8984): Gtk-WARNING **: 21:49:10.381: Could not load a pixbuf from icon theme.
This may indicate that pixbuf loaders or the mime database could not be found.
**
Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /org/gtk/libgtk/icons/48x48/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Bail out! Gtk:ERROR:../../../../gtk/gtkiconhelper.c:494:ensure_surface_for_gicon: assertion failed (error == NULL): Failed to load /org/gtk/libgtk/icons/48x48/status/image-missing.png: Unrecognized image file format (gdk-pixbuf-error-quark, 3)
Aborted
```

**Comment by m4dEngi** [comment]
It should not be used with sudo. What version of steam client are you using? (how did you install the steam client)

**Comment by mooo25** [comment]
I am using Version 1.0.0.78.  I checked, and it is up to date.

I installed the Flathub using the Mint Software Manager.

I hope I am answering your question.  Def a noob at Linux.

**Comment by m4dEngi** [comment]
The app can't work with Flatpak/Snap version of steam as it needs access to running client process for IPC calls.

**Comment by m4dEngi** [comment]
Latest appimage should work better, but still requires non flatpak/snap native version of steam client.

**Comment by mooo25** [comment]
Thank you!

So reinstalling Steam using the non flatpak version allowed the AppImage to
start.

We ran into a problem using the program though.  I was able to send the
invite to my friend from the appimage, but when he clicked on the
invitation, he was given a choice to use the Steam Link App or the Steam
App.  This choice was given to him from the website he was brought to.

I believe no matter which app he chose, he ran into an error each time, I
think it was "Unable to get stream".  It also locked up his computer so
that he had to restart it every attempt.

**Comment by m4dEngi** [comment]
The part after sending the invite is handled by Steam itself and is outside the scope of RPW. Unfortunately, I can't help with that, as it seems there is some connectivity/steam/system issue on your friend's end that prevents steam from connecting to the remote session.

**Comment by JJ-8** [comment]
> Latest appimage should work better, but still requires non flatpak/snap native version of steam client.

The latest appimage at the moment does work with at least snap (probably also flatpak), but you should add the .AppImage file as a new non-steam game to your library and then run it from there in order to make it work.
